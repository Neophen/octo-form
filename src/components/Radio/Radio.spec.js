import { shallowMount } from "@vue/test-utils";
import ORadio from "./Radio";

let wrapper;

describe("ORadio", () => {
  beforeEach(() => {
    wrapper = shallowMount(ORadio);
  });

  it("is called", () => {
    expect(wrapper.name()).toBe("ORadio");
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("has an input radio", () => {
    expect(wrapper.contains("label input[type=radio]")).toBeTruthy();
  });

  it("emit input event with value when value change", done => {
    wrapper.setProps({ value: true });
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.computedValue).toBeTruthy();
      wrapper.vm.computedValue = false;
      const valueEmitted = wrapper.emitted()["input"][0];
      expect(valueEmitted).toContainEqual(false);
      done();
    });
  });

  it("method focus() gives focus to the input element", done => {
    wrapper.vm.$refs.input.focus = jest.fn();
    wrapper.vm.focus();
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$refs.input.focus).toHaveBeenCalled();
      done();
    });
  });
});
