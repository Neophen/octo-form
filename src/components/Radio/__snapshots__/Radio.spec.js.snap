// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ORadio render correctly 1`] = `<label class="octo-radio"><input type="radio" class="octo-radio__input" value="false"> <span class="octo-radio__mark"></span> <span class="octo-radio__label"></span></label>`;
